Tools/autotest/arduplane.py

```diff
@@ -4145,6 +4145,37 @@ def MANUAL_CONTROL(self):
             0) # button mask
         self.fly_home_land_and_disarm()
 
+    def AltResetBadGPS(self):
+        '''Tests the handling of poor GPS lock pre-arm alt resets'''
+        self.set_parameters({
+            "SIM_GPS_GLITCH_Z": 0,
+            "SIM_GPS_ACC": 0.3,
+        })
+        self.wait_ready_to_arm()
+
+        m = self.mav.recv_match(type='GLOBAL_POSITION_INT', blocking=True)
```

Comment:
`recv_match` like this was eliminated - it will block forever in the case something has seriously gone wrong with the vehicle.  Not going to happen here, but it's a pattern thing.

```suggestion
        m = self.assert_receive_message('GLOBAL_POSITION_INT')
```

You could also `get_altitude(relative=True)` which returns the height in metres - but I'm not sure that's the right thing to do here in that I think we really do want to know exactly where this altitude is coming from.

Reply:
done